doc/CastingTypes.md

```diff
@@ -0,0 +1,117 @@
+# Casting Types
+
+Using c-casting style must not be used within the code implementation. Below type-castings can be found in the SDK code which have been reviewed and approved as safe convertions.
+
+All type-casting statements are expected to include a comment to this file and the types should been verify and included to this list.
+
+## static cast
+
+## dynamic cast
+
+## const cast
+
+## reinterpret cast
```

Comment:
@CaseyCarter  can you help me to review below list of type castings and mention if there is any of them which should not be done and re-designed?  (specially for the ones as `ToBeConfirmed`